tao_compiler/mlir/disc/transforms/lower_tf.cc

```diff
@@ -643,6 +651,30 @@ LogicalResult ConvertDequantizeOp::dequantize(TF::DequantizeOp op,
                          << "\n";
 }
 
+struct ConvertFakeQuantOp : public OpRewritePattern<TF::DiscFakeQuantOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TF::DiscFakeQuantOp op,
+                                PatternRewriter& rewriter) const final {
+    auto loc = op.getLoc();
+    auto input = op.input();
+    auto scale = op.scale();
+    auto zero_point = op.zero_point();
+    SmallVector<int64_t> axis;
+    for (auto& v : op.axis()) {
+      axis.push_back(v.cast<IntegerAttr>().getInt());
+    }
+    Value new_op = rewriter.create<mhlo_disc::FakeQuantOp>(
+        loc, input.getType(), input, scale, zero_point, op.use_signedAttr(),
```

Comment:
why use input type instead of `op.getType()` as the result type of the new op.

Reply:
done.